File: src/test/java/com/dotnomi/fabricdependencyinjection/java/testclasses/identifier/FileStorage.java
package com.dotnomi.fabricdependencyinjection.java.testclasses.identifier;

import com.dotnomi.fabricdependencyinjection.annotation.ModIdentifier;
import com.dotnomi.fabricdependencyinjection.annotation.ModScoped;

@ModScoped
@ModIdentifier("file")
public class FileStorage implements StorageService {
  @Override
  public String store(String data) {
    return "Saved to file";
  }
}

File: src/test/java/com/dotnomi/fabricdependencyinjection/kotlin/testclasses/identifier/FileStorage.kt
package com.dotnomi.fabricdependencyinjection.kotlin.testclasses.identifier

import com.dotnomi.fabricdependencyinjection.annotation.ModIdentifier
import com.dotnomi.fabricdependencyinjection.annotation.ModScoped

@ModScoped
@ModIdentifier("file")
class FileStorage : StorageService {
    override fun store(data: String): String {
        return "Saved to file"
    }
}
